arc/database/api.py

```
@@ -741,6 +759,121 @@ def _categorize_error(self, error_message: Optional[str]) -> Optional[str]:
         else:
             return "system_error"  # Default fallback for any other errors
 
+    async def record_batch_tool_usage(
+        self,
+        tool_usage_records: List[Dict[str, Any]]
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)